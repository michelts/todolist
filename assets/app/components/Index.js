import React from 'react';
import { HashRouter as Router, Switch, Route } from 'react-router-dom';
import Container from 'common/components/Container';
import Header from './Header';
import Login from './Login';
import Logout from './Logout';
import Register from './Register';
import Tasks from './Tasks';

const Index = () => (
  <Router>
    <Header>
      Todo list application
    </Header>
    <Container>
      <Switch>
        <Route path="/tasks">
          <Tasks />
        </Route>
        <Route path="/register">
          <Register />
        </Route>
        <Route path="/logout">
          <Logout />
        </Route>
        <Route>
          <Login />
        </Route>
      </Switch>
    </Container>
  </Router>
);

export default Index;
