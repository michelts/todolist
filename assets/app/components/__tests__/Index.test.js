import React from 'react';
import { shallow } from 'enzyme';
import Index from '../Index';

jest.unmock('react-router-dom');

describe('Index component', () => {
  it('should render routes to Login, Logout, Register and Tasks', () => {
    const wrapper = shallow(<Index />);
    expect(wrapper).toMatchSnapshot();
  });
});
