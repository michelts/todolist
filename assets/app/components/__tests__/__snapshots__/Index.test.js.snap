// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index component should render routes to Login, Logout, Register and Tasks 1`] = `
<HashRouter>
  <Header>
    Todo list application
  </Header>
  <Container>
    <Switch>
      <Route
        path="/tasks"
      >
        <Tasks />
      </Route>
      <Route
        path="/register"
      >
        <Register />
      </Route>
      <Route
        path="/logout"
      >
        <Logout />
      </Route>
      <Route>
        <Login />
      </Route>
    </Switch>
  </Container>
</HashRouter>
`;
